django/utils/regex_helper.py

```diff
@@ -344,8 +344,7 @@ def _compile():
         if isinstance(regex, (str, bytes)):
             return re.compile(regex, flags)
         else:
-            assert not flags, (
-                'flags must be empty if regex is passed pre-compiled'
-            )
+            if flags:
+                raise ValueError('Flags must be empty if regex is passed pre-compiled.')
```

Comment:
We decided that there is no need to change this, see https://github.com/django/django/pull/14114#discussion_r592670434.